### PR TITLE
<feat>: Add android logcat support.

### PR DESCRIPTION
**Pull request by xbwee1024**
debug log will disapear if it's called in android app, so this PR add logcat instead of printf.

### CODE REVIEW

File: whisper.cpp [hunk 1]
```diff
@@ -118,6 +118,9 @@ static void byteswap_tensor(ggml_tensor * tensor) {
 // logging
 //
 
+#ifdef WHISPER_LOGCAT
+#include <android/log.h>
+#endif
```
Comment (bobqianic) [inline whisper.cpp:123]: To be honest, I'm not convinced that including `<android/log.h>` in whisper.cpp is the best approach. It seems more practical to use a callback instead. 

I'm also wondering why do we continue to use `WHISPER_PRINT_DEBUG` when we already have `WHISPER_LOG`? Couldn't we just create a function named `WHISPER_LOG_DEBUG` that offers both console output and callback functionality? @ggerganov

Comment (xbwee1024) [reply whisper.cpp:123]: Well, I see the callback logic just now,  I agree with you, it's the right way!

Comment (ggerganov) [reply whisper.cpp:123]: Yup, logcat is better to not be part of `whisper.cpp` - utilize the log callback

Regarding `WHISPER_PRINT_DEBUG` - it's used for debugging that I don't think will be really useful to the "outside world". But we could add `WHISPER_LOG_DEBUG` instead for consistency. Probably have to extend `enum ggml_log_level` with new `GGML_LOG_LEVEL_DEBUG` value

**Comment by bobqianic** [comment]
I am now closing this PR because you can now use a callback to get all the log information, including DEBUG.

**Pull request closed without merging**